src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -924,16 +857,228 @@ export class SearchSource {
     // });
     // Alternatively you could also add this query via "Edit as Query DSL", then it needs no code to be changed
 
-    if (highlightAll && body.query) {
-      body.highlight = getHighlightRequest(getConfig(UI_SETTINGS.DOC_HIGHLIGHT));
-      delete searchRequest.highlightAll;
+    body._source = _source;
+
+    // only include unique values
+    if (sourceFieldsProvided) {
+      if (!isEqual(remainingFields, fieldsFromSource)) {
+        setWith(body, '_source.includes', remainingFields, (nsValue) => {
```

Comment:
This is driving me crazy - this is object mutation code that I'd like to rewrite without object mutation but I'm having a very hard time figuring out how to do this.

Reply:
I wonder if this is all extraneous and could just be replaced with

```ts
    if (sourceFieldsProvided && !isEqual(remainingFields, fieldsFromSource)) {
      body._source.includes = remainingFields;
    }
```